packages/components/src/toggle-group-control/toggle-group-control/component.tsx

```diff
@@ -20,43 +20,75 @@ import { VisualLabelWrapper } from './styles';
 import { useOnValueUpdate } from '../../utils/hooks/use-on-value-update';
 import { useEvent, useMergeRefs } from '@wordpress/compose';
 
 /**
- * A utility used to animate something (e.g. an indicator for the selected option
- * of a component).
+ * A utility used to animate something in a container component based on the "offset
+ * rect" (position relative to the container and size) of a subelement. For example,
+ * this is useful to render an indicator for the selected option of a component, and
+ * to animate it when the selected option changes.
  *
- * It works by tracking the position and size (i.e., the "rect") of a given subelement,
- * typically the one that corresponds to the selected option, relative to its offset
- * parent. Then it:
+ * Takes in a container element and the up-to-date "offset rect" of the target
+ * subelement, obtained with `useTrackElementOffsetRect`. Then it does the following:
  *
- * - Keeps CSS variables with that information in the parent, so that the animation
- *   can be implemented with them.
- * - Adds a `is-animation-enabled` CSS class when the element changes, so that the
- *   target (e.g. the indicator) can be animated to its new position.
- * - Removes the `is-animation-enabled` class when the animation is done.
+ * - Adds CSS variables with rect information to the container, so that the indicator
+ *   can be rendered and animated with them. These are kept up-to-date, enabling CSS
+ *   transitions on change.
+ * - Sets an attribute (`data-subelement-animated` by default) when the tracked
+ *   element changes, so that the target (e.g. the indicator) can be animated to its
+ *   new size and position.
+ * - Removes the attribute when the animation is done.
+ *
+ * The need for the attribute is due to the fact that the rect might update in
+ * situations other than when the tracked element changes, e.g. the tracked element
+ * might be resized. In such cases, there is no need to animate the indicator, and
+ * the change in size or position of the indicator needs to be reflected immediately.
  */
-function useSubelementAnimation(
-	subelement?: HTMLElement | null,
+function useAnimatedOffsetRect(
+	/**
+	 * The container element.
+	 */
+	container: HTMLElement | undefined,
+	/**
+	 * The rect of the tracked element.
+	 */
+	rect: ElementOffsetRect,
 	{
-		parent = subelement?.offsetParent as HTMLElement | null | undefined,
 		prefix = 'subelement',
+		attribute = `${ prefix }-animated`,
 		transitionEndFilter,
 	}: {
-		parent?: HTMLElement | null | undefined;
+		/**
+		 * The prefix used for the CSS variables, e.g. if `prefix` is `selected`, the
+		 * CSS variables will be `--selected-top`, `--selected-left`, etc.
+		 * @default 'subelement'
+		 */
 		prefix?: string;
+		/**
+		 * The name of the data attribute used to indicate that the animation is in
+		 * progress.
+		 * @default `${ prefix }-animated`
+		 */
+		attribute?: string;
```

Comment:
Should we call this `dataAttribute` ? Or keep it `attribute` but expect that the `data-` part is included in the variable?

Reply:
Eh, I don't think it changes much either way. I'll rename if you prefer, but don't really see the point of it. It's just a way to customize the attribute, which happens to be a data attribute. The current name seems appropriate to me as is.

Reply:
As a developer, when I see `attribute`  I expect to be specifying the full attribute name, and I wouldn't expect `data-` to be prefixed. I would then be confused when having to write styles that rely on such HTML attribute.

It would be a small change and at no cost from our side, and it would help clarify the situation for future consumers of the function

Reply:
Renamed to `dataAttribute`. I wanna make sure that `data-` is added, so I think it's the better route while addressing your concerns. Also improved the docs to ensure clarity 👍